core/src/main/java/io/grpc/ServerInterceptors.java

```diff
@@ -80,6 +84,50 @@ public static ServerServiceDefinition intercept(ServerServiceDefinition serviceD
     return serviceDefBuilder.build();
   }
 
+  /**
+   * Create a new {@code ServerServiceDefinition} whose {@link MethodDescriptor} serializes to
+   * and from byte[] for all methods.  The {@code ServerCallHandler} created will automatically
+   * convert back to the original types for request and response before calling the existing
+   * {@code ServerCallHandler}.  Calling this method combined with the intercept methods will
+   * allow the developer to choose whether to intercept messages of byte[], or the modeled types
+   * of their application.
+   *
+   * @param serviceDef the service definition to convert messages to byte[]
+   * @return a wrapped version of {@code serviceDef} with the byte[] conversion applied.
+   */
+  public static ServerServiceDefinition useByteArrayMessages(
+      final ServerServiceDefinition serviceDef) {
+    final MethodDescriptor.Marshaller<byte[]> marshaller =
```

Comment:
How do you feel about using `InputStream` instead of `byte[]`? With `InputStream` it's possible to do efficient pass-through and make use of any additional optimization features we include in `InputStream` (such as Drainable, but we may add more like ability to byte[][] or the ByteBuffers, etc). It seems it is relatively easy (although slightly annoying) to convert from `InputStream` to byte[], so maybe it won't add too much burden?